metricbeat/module/system/socket_summary/socket_summary.go

```diff
@@ -45,12 +46,15 @@ func init() {
 type MetricSet struct {
 	mb.BaseMetricSet
 	sockstat string
+	mod      system.Resolver
 }
 
 // New creates a new instance of the MetricSet. New is responsible for unpacking
 // any MetricSet specific configuration options if there are any.
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	sys := base.Module().(system.Resolver)
```

Comment:
Should we check the conversion here just in case?

Reply:
The resolver implementation is basically hard-coded, so I'm not sure what edge cases that would cover?

Reply:
I was thinking on future refactors, but I guess this would be detected by panics on tests.